corehq/apps/hqcase/export.py

```diff
@@ -56,7 +60,7 @@ def get_matching_owner(case):
             return get_owner_id(case)
 
     def might_be_relevant(case):
-        return not filter_group or get_owner_id(case) in owner_ids
+        return get_owner_id(case) in owner_ids
```

Comment:
Based on testing and my reading of the code this should return all the unknown user data if unknown users is selected. The function that pulls the users should pull any user that has submitted something to the domain. I tested that it works even after kicking a web user out of the domain.